### PR TITLE
Configure primary nic if only set in cfg file

### PR DESCRIPTION
**Pull request by ChaitanyaKulkarni28**
This is follow up PR for #402 to make sure use-domains is set as expected and primary nic configuration is dependent on config value set in the file

/cc @a-crate @drewhli

### CODE REVIEW

File: google_guest_agent/network/manager/netplan_linux.go [hunk 1]
```diff
@@ -274,6 +277,16 @@ func (nd networkdNetplanDropin) write(n netplan, iface string) error {
 	return nil
 }
 
+// shouldUseDomains returns true if interface index is 0.
+func shouldUseDomains(idx int) *bool {
+	trueVal := true
```
Comment (a-crate) [inline google_guest_agent/network/manager/netplan_linux.go:282]: This can be simpler, can we just do something like `val := idx ==0; return &val`?

Comment (ChaitanyaKulkarni28) [reply google_guest_agent/network/manager/netplan_linux.go]: Done

**Review by drewhli — APPROVED** [review]

**Pull request merged**